SingularityService/src/main/java/com/hubspot/singularity/data/history/SingularityDeployHistoryPersister.java

```diff
@@ -56,16 +59,22 @@ public void runActionOnPoll() {
 
       Optional<SingularityDeployHistory> deployHistory = deployManager.getDeployHistory(deployKey.getRequestId(), deployKey.getDeployId(), true);
 
-      if (!deployHistory.isPresent()) {
+      if (deployHistory.isPresent()) {
+        deployHistoryByRequestId.put(deployKey.getRequestId(), deployHistory.get());
+      } else {
         LOG.info("Deploy history for key {} not found", deployKey);
-        continue;
       }
+    }
 
-      if (moveToHistoryOrCheckForPurge(deployHistory.get())) {
-        numTransferred++;
-      }
+    for (Collection<SingularityDeployHistory> deployHistoryForRequest : deployHistoryByRequestId.asMap().values()) {
+      int i=0;
+      for (SingularityDeployHistory deployHistory : deployHistoryForRequest) {
```

Comment:
Do we need to sort the `Collection<SingularityDeployHistory>` first here? I saw where we were sorting for tasks and for requests, but did not see any sort for deploy history.